### ISSUE TITLE
Flightmaster displayed discounted prices do not match actual deducted amount

### ISSUE THREAD

**Issue by wrathofthebug** [dalaranwow/dalaran-wow#4923]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
Flightmaster prices displayed are accurate but the amount deducted after choosing the flight destination is the original non-discounted amount.
**Expected behaviour**: Tell us what should happen instead
The amount deducted should match the discounted amount based on reputation
**Steps to reproduce the problem**:

1. Go to a flightmaster whose race/faction you have friendly or better reputation with
2. Hover over any of the possible flight destinations and note the price
3. Before confirming flight destination check your current gold and after confirming check your gold again and calculate the amount deducted to see if it matches the discounted amount displayed on the flightmaster's map.

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by TeridanDW** [comment]
Hello and thank you for the report @wrathofthebug.
I can confirm the issue on PTR.

**Comment by TeridanDW** [comment]
Works as intended on PTR if player doesn't relog while in flight.
If player use multyflight path and relog while he is in air on login and at nextflight master he will pay the full price instead of the reduced by rep one.

**Comment by robinsch** [comment]
I already have a fix for this, will be pushed to PTR next days.

**Comment by robinsch** [comment]
Updated.

**Comment by TeridanDW** [comment]
Works as intended on PTR.

**Comment by wrathofthebug** [comment]
Hi it's been almost 5 months since the last update. May I know if the fix has been push out to Algalon server?
